src/core/io_queue.cc

```diff
@@ -683,7 +683,16 @@ io_group::io_group(io_queue::config io_cfg, unsigned nr_queues)
         const auto& fg = _fgs[g_idx];
         auto max_cap = fg.maximum_capacity();
         for (unsigned shift = 0; ; shift++) {
-            auto tokens = internal::request_tokens(io_direction_and_length(idx, 1 << (shift + io_queue::block_size_shift)), _config);
+            // Break this loop if the calculated request length gets larger than the
+            // request length limit. This avoids an infinite loop if the
+            // configured io_properties are huge and capacity ends up being always 0.
+            // _max_request_length will hold default values of io_group::request_length_limit
+            unsigned long req_length = 1ul << (shift + io_queue::block_size_shift);
+            if (req_length > io_group::request_length_limit) {
+                break;
```

Comment:
Need an info-level message here saying that maximum request length limit was cut off early at some "default" value